### PR TITLE
chore: Add tracking for hyperlinks

### PR DESCRIPTION
**Pull request by ad1992**

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -2687,6 +2687,7 @@ class App extends React.Component<AppProps, AppState> {
       !this.state.selectedElementIds[this.hitLinkElement.id]
     ) {
       setCursor(this.canvas, CURSOR_TYPE.POINTER);
+
```
Comment (ad1992) [inline src/components/App.tsx:2690]:
```suggestion
```

File: src/element/Hyperlink.tsx [hunk 2]
```diff
@@ -69,6 +70,10 @@ export const Hyperlink = ({
 
     const link = normalizeLink(inputRef.current.value);
 
+    if (!element.link && link) {
+      trackEvent("hyperlink", "create");
+    }
```
Comment (dwelle) [inline src/element/Hyperlink.tsx:75]: I moved the creation in here and only checking if we actually submit the form with non-empty link (from previously empty)

File: src/element/Hyperlink.tsx [hunk 3]
```diff
@@ -109,7 +113,7 @@ export const Hyperlink = ({
   }, [appState, element, isEditing, setAppState]);
 
   const handleRemove = useCallback(() => {
-    trackEvent("link", "Delete");
+    trackEvent("hyperlink", "delete");
```
Comment (dwelle) [inline src/element/Hyperlink.tsx:116]: aligning the casing (you mixed uppercase with lwoercase)

File: src/element/Hyperlink.tsx [hunk 4]
```diff
@@ -118,7 +122,7 @@ export const Hyperlink = ({
   }, [setAppState, element, isEditing]);
 
   const onEdit = () => {
-    trackEvent("link", "Update");
+    trackEvent("hyperlink", "edit");
```
Comment (dwelle) [inline src/element/Hyperlink.tsx:125]: let's distinguish between an `update` and `edit`. In here we're tracking when a user clicks on the edit button, but doesn't necessarily update. 

I didn't add a separate `update` event as I don't think it's useful knowledge. Technically even the edit isn't, if we don't distinguish how the user edits (i.e. which button they used), so we wouldn't be able to figure out which UI is being used more often.

We should prob remove this altogether or track the actual UI button.

File: src/element/Hyperlink.tsx [hunk 5]
```diff
@@ -244,11 +248,8 @@ export const actionLink = register({
   name: "link",
   perform: (elements, appState) => {
     if (appState.showHyperlinkPopup === "editor") {
-      trackEvent("link", "close");
-
```
Comment (dwelle) [inline src/element/Hyperlink.tsx]: IMO not useful to know

File: src/actions/manager.tsx [hunk 6]
```diff
@@ -10,6 +10,31 @@ import {
...
+import { trackEvent } from "../analytics";
+
+const trackAction = (
+  action: Action,
+  source: "ui" | "keyboard" | "api",
+  value: any,
+) => {
+  if (action.trackEvent !== false) {
+    try {
+      if (action.trackEvent === true) {
```
Comment (ad1992) [inline src/actions/manager.tsx:22]: when is the boolean case used ?

Comment (dwelle) [reply src/actions/manager.tsx:22]: not used currently. Will be used once this lands to master and I'll issue the second PR.

**Review by dwelle — APPROVED** [review]

**Pull request merged**